### ISSUE TITLE
Build & test the windows installer

### ISSUE THREAD

**Issue by Symbolics** [Symbolics/scl#2]
Like the name says...

We should offer to contribute the installer to the SBCL folks so that people can use an up-to-date version.  No one seems to be maintaining the installer.

**Comment by snunez1** [comment]
@beberman, is this something you have time to look at?  I thought you mentioned MS Windows as your daily driver.  If that's not the case I can take this one and perhaps you can look at the CI & ANSI test setup?

**Comment by beberman** [comment]
I usually use SBCL under the ubuntu that can be installed on windows 10.  I just tried the cygwin build under direct windows 10. Hit quite a few compile errors mostly from signal.h and pthreads_win32.h.   I can track those down and patch but would like to confirm you agree with that first.  

Sample bug:
pthreads_win32.h:48:16: error: redefinition of 'struct timespec'
   48 | typedef struct timespec {
      |                ^~~~~~~~
In file included from /usr/include/sys/timespec.h:38,
                 from /usr/include/sys/select.h:16,
                 from /usr/include/sys/types.h:50,
                 from /usr/include/time.h:28,
                 from pthreads_win32.h:4,
                 from runtime.h:21,
                 from os.h:21,
                 from interr.h:15,
                 from gc-assert.h:15,
                 from gc.h:20,
                 from alloc.c:18:

**Comment by snunez1** [comment]
Can you provide the command used to produce this error?  I have built SBCL, using itself as the initial compiler, quite a few times and don't recall any C files being rebuilt.  I suppose it's worth mentioning that we're not talking about SBCL under WSL; this is a 'native' MS Windows install.

**Comment by beberman** [comment]
Sure this is a cygwin build on windows10.   Just did:
> sh make.sh

command that fails is:
gcc -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -g -W -Wall -Wno-unused-function -Wno-unused-parameter -Wno-cast-function-type -Wno-type-limits -fno-omit-frame-pointer -O5 -m64 -DWINVER=0x0501 -D__W32API_USE_DLLIMPORT__ -fno-pie -I.  -c -o alloc.o alloc.c

**Comment by snunez1** [comment]
Are you using the 'correct' MSYS2 shell?  I am able to compile 2.3.8 from source with `sh make.sh` command using the `MINGW64` shell.  You have to be careful which MSYS2 shell you use for a particular purpose because they are all set up for different things.

**Comment by beberman** [comment]
As I indicated I used cygwin that's different from msy2.  Cygwin is supposed to be posix where mysy2 is windows.  SBCL should also compile under cygwin.

**Comment by snunez1** [comment]
Ah, right.  We might need to see help from the SBCL mailing list.  I don't think cygwin and MSYS2 can co-exist on a single machine.  MSYS2 is a derivative of cygwin and is also posix.

Alternatively we can try to debug cygwin.  Either way though, I think asking/involving the upstream maintainers is a good idea, since if we patch for cygwin we'll need them to accept the patches upstream.  They might not want that, in which case it may be better to use MSYS2 (which also has the advantage of us both being on the same platform me being able to use the same codebase).

**Comment by snunez1** [comment]
I think it's going to have to be MSYS2.  See https://bugs.launchpad.net/sbcl/+bug/1720642 where one of the SBCL developers says cygwin is no longer supported.

**Comment by beberman** [comment]
I'll switch to MSYS2 this week.  Those are the errors I got. They should at least change the make.sh to spit out "NOT SUPPORTED" at the very beginning when it detects cygwin.  Should we make that change?

**Comment by snunez1** [comment]
> I'll switch to MSYS2 this week. Those are the errors I got. They should at least change the make.sh to spit out "NOT SUPPORTED" at the very beginning when it detects cygwin. Should we make that change?

Good idea.  Let's submit that upstream too; it will also raise awareness that someone is working on the MS Windows version.

**Comment by snunez1** [comment]
Just FYI, it looks like there's a rapid commit pace upstream, so we'll need to rebase before submitting changes.  With the current approach of just _adding_ to SBCL, a rebase should be trivial, but something to be aware of.

**Comment by beberman** [comment]
ok I can build now on msys2 and run the sh install.sh.  You want to get the windows installer to work?

**Comment by snunez1** [comment]
I just assigned myself.  Work commitments mean I won't get to this for another week or two, and if you want to take it on, please do.  At this stage we just need to _build_ the installer.  Once we can do that we can look at the additions of the logical pathnames and how to fold in MAGMA/MKL/Cephes.
